zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -870,6 +874,73 @@ impl<F: Field> MPTConfig<F> {
+                    offset += 1;
+                }
+                for ind in 184..=191 {
+                    // very long string
+                    assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::RLP.scalar())?;
+                    assignf!(region, (self.fixed_table[1], offset) => ind.scalar())?;
+                    assignf!(region, (self.fixed_table[2], offset) => true.scalar())?;
+                    assignf!(region, (self.fixed_table[3], offset) => false.scalar())?;
+                    assignf!(region, (self.fixed_table[4], offset) => true.scalar())?;
+                    offset += 1;
+                }
+                for ind in 192..=247 {
+                    // short list
+                    assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::RLP.scalar())?;
+                    assignf!(region, (self.fixed_table[1], offset) => ind.scalar())?;
+                    assignf!(region, (self.fixed_table[2], offset) => false.scalar())?;
+                    assignf!(region, (self.fixed_table[3], offset) => true.scalar())?;
+                    assignf!(region, (self.fixed_table[4], offset) => false.scalar())?;
+                    offset += 1;
+                }
+                // 248
+                // long list
+                assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::RLP.scalar())?;
+                assignf!(region, (self.fixed_table[1], offset) => 248i32.scalar())?;
+                assignf!(region, (self.fixed_table[2], offset) => false.scalar())?;
+                assignf!(region, (self.fixed_table[3], offset) => false.scalar())?;
+                assignf!(region, (self.fixed_table[4], offset) => false.scalar())?;
+                offset += 1;
+                // 249
+                // very long list
+                assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::RLP.scalar())?;
+                assignf!(region, (self.fixed_table[1], offset) => 249i32.scalar())?;
+                assignf!(region, (self.fixed_table[2], offset) => false.scalar())?;
+                assignf!(region, (self.fixed_table[3], offset) => false.scalar())?;
+                assignf!(region, (self.fixed_table[4], offset) => true.scalar())?;
+                offset += 1;
+
+                // KEY_PREFIX_ODD: u8 = 0b0001_0000;
+                // first byte of ext key >> 4 == 1
+                for ind in 0..256 {
+                    assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::ExtOddKey.scalar())?;
+                    assignf!(region, (self.fixed_table[1], offset) => ind.scalar())?;
+                    if 16 <= ind && ind < 32 {
+                        assignf!(region, (self.fixed_table[2], offset) => true.scalar())?;
+                    } else {
+                        assignf!(region, (self.fixed_table[2], offset) => false.scalar())?;
+                    }
+                    offset += 1;
+                }
```

Comment:
This works as expected, but I checked the compact encoding spec again and it seems like to enforce the correct encoding we can limit the possible values so the circuit has to do less of that:
```
// Even - only the nibbles 0 0 are valid
assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::ExtOddKey.scalar())?;
assignf!(region, (self.fixed_table[1], offset) => 0.scalar())?;
assignf!(region, (self.fixed_table[2], offset) => false.scalar())?;
offset += 1;
// Odd - First nibble is 1, the second nibble can be any value
for idx in 0..16 {
    assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::ExtOddKey.scalar())?;
    assignf!(region, (self.fixed_table[1], offset) => ((0b1_0000) + idx).scalar())?;
    assignf!(region, (self.fixed_table[2], offset) => true.scalar())?;
    offset += 1;
}

Reply:
Yea you're right, I saw this [https://github.com/privacy-scaling-explorations/zkevm-specs/blob/21640e4834a5eb36fc2fb4f4108520d1c4f4f655/specs/mpt/extension-node.md?plain=1#L237](url).
Will make the change 👌